### PR TITLE
Bump EvalEx to 3.0.5.

### PR DESCRIPTION
**Pull request by FelixOBrien**
Addresses #280 
- Updated EvalEx dependency in Gradle
- Rewrote all EvalEx functions

### CODE REVIEW

File: core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+import com.ezylang.evalex.parser.ParseException;
+import org.junit.jupiter.api.Test;
+
+import java.math.BigDecimal;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+public class ConditionFunctionTest extends FactoryTest {
+    ExpressionConfiguration config = ExpressionConfiguration.defaultConfiguration().withAdditionalFunctions(
```
Comment (virustotalop) [inline core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java:18]: Remove this and wrap each test in a condition function instead. So code will work something like this
```java
ConditionFunction function = new ConditionFunction();
function.setData("expressionString");
assertTrue/assertFalse(function.function(this.getFactory().createPlayer());
```

File: core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+
+        conditionFunction.setData("STRCONTAINS(\"test\", \"es\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("AFTERDATE(\"01 01, 2023 00:00:00\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+
+        conditionFunction.setData("BEFOREDATE(\"01 01, 2100 00:00:00\")");
+        assertTrue(conditionFunction.function(playerWrapper));
+    }
```
Comment (virustotalop) [inline core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java:183]: Can these be separated into smaller units I.E: `testStringEqualsTrue` `testStringEqualsFalse` `testStringEqualsEmpty` etc? The goal here is to have each have one small unit for each so we can catch edge cases easier without having to dig through the code.

**Review by virustotalop — CHANGES_REQUESTED** [review]

File: core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java (hunk 1 above)
Comment (FelixOBrien) [reply core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java:18]: Will do.

File: core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java (hunk 2 above)
Comment (FelixOBrien) [reply core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java:183]: Good point. I'll split them up.

**Review by virustotalop — APPROVED** [review]

File: core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java (hunk 1 above)
Comment (FelixOBrien) [reply core/src/test/java/com/clubobsidian/dynamicgui/core/test/function/ConditionFunctionTest.java]: Addressed in latest commit.
